modules/core/src/v2/coins/xlm.ts

```diff
@@ -459,52 +460,50 @@ export class Xlm extends BaseCoin {
   }
 
   /**
-   * Attempt to resolve a stellar address into a stellar account
-   * If address domain matches bitgo's then resolve on our federation server
-   * Else, make the request to the federation server hosting the address
+   * Perform federation lookups
+   * Our federation server handles lookups for bitgo as well as for other federation domains
    *
-   * @param address - stellar address to look for
+   * @param {String} [address] - address to look up
+   * @param {String} [accountId] - account id to look up
    */
-  federationLookupByName(address: string): Bluebird<stellar.FederationServer.Record> {
+  federationLookup({ address, accountId }: { address?: string, accountId?: string }): Bluebird<stellar.FederationServer.Record> {
     const self = this;
     return co<stellar.FederationServer.Record>(function *() {
-      const addressParts = _.split(address, '*');
-      if (addressParts.length !== 2) {
-        throw new InvalidAddressError(`invalid stellar address: ${address}`);
-      }
-      const [, homeDomain] = addressParts;
       try {
-        if (homeDomain === self.homeDomain) {
-          const federationServer = self.getBitGoFederationServer();
+        const federationServer = self.getBitGoFederationServer() as stellar.FederationServer;
+        if (address) {
           return yield federationServer.resolveAddress(address);
-        } else {
-          return yield stellar.FederationServer.resolve(address);
+        } else if (accountId) {
+          return yield federationServer.resolveAccountId(accountId);
         }
       } catch (e) {
-        if (e.message === 'Network Error') {
-          throw e;
-        } else {
-          throw new Error('account not found');
-        }
+        throw new StellarFederationUserNotFoundError(_.get(e, 'response.data.detail'));
       }
     }).call(this);
   }
 
+  /**
+   * Attempt to resolve a stellar address into a stellar account
+   *
+   * @param {String} address - stellar address to look for
+   */
+  federationLookupByName(address: string): Bluebird<stellar.FederationServer.Record> {
```

Comment:
Should this (and perhaps `federationLookupById`) be made private now that we have `federationLookup`, which can handle both types? Seems like either one or the other should be made private.

Reply:
`federationLookupByName` and `federationLookupById` should be public. I made `federationLookup` private.